.github/workflows/test.yml

```diff
@@ -98,17 +112,20 @@ jobs:
           - dev
     steps:
       - uses: actions/checkout@v4
+        if: contains(needs.check_changes.outputs.path_filters, 'docker')
       - name: publish images
         if: contains(needs.check_changes.outputs.path_filters, 'docker')
```

Comment:
duplicated lines?

Reply:
No, just applying the conditional to the checkout too. Just trying to make it faster if we don't need to build the images.

Can't skip the whole job though because there's not really a way to have other jobs
1. run fine if this job is skipped, AND
2. wait for this job to run if it's not skipped.

As far as I can tell